### PR TITLE
NIAD-1407: Removed assumption of resource type

### PR DESCRIPTION
**Pull request by ElliotB-96**
- Removed workaround for resource type assumption
- Updated wiremock test for e2e testing

### CODE REVIEW

**Review by msolecka — DISMISSED** [review]

**Review by andrewwa-kainos — APPROVED** [review]

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementExtractor.java [hunk 1]
```diff
@@ -125,12 +125,6 @@ public static String extractStatusReasonStoppedAvailabilityTime(MedicationReques
     public static String extractIdFromPlanMedicationRequestReference(Reference reference, MessageContext messageContext) {
         LOGGER.debug("Ensuring the bundle contains Plan MedicationRequest {}", reference.getReference());
 
-        if (reference.getReferenceElement().getResourceType() == null) {
-            // TODO: workaround for NIAD-1407 the type should never be assumed
-            LOGGER.warn("Reference {} is missing a resource type. Assuming MedicationRequest resource type.", reference.getReference());
-            reference = buildReference(ResourceType.MedicationRequest, reference.getReferenceElement().getIdPart());
```
Comment (matthewdoreilly) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementExtractor.java:131]: I think this `buildReference` method can be deleted as well?

**Review by msolecka — APPROVED** [review]

File: docker/wiremock/stubs/__files/correctPatientNoDocsStructuredRecordResponse.json [hunk 2]
```diff
@@ -243,7 +244,7 @@
                         {
                             "system": "http://snomed.info/sct",
                             "code": "886921000000105",
-                            "display": "Active Allergies"
+                            "display": "Allergies and adverse reactions"
```
Comment (matthewdoreilly) [inline docker/wiremock/stubs/__files/correctPatientNoDocsStructuredRecordResponse.json:247]: Just curious, why this change?

**Review by matthewdoreilly — APPROVED** [review]

File: docker/wiremock/stubs/__files/correctPatientNoDocsStructuredRecordResponse.json (hunk 2 above)
Comment (mattd-kainos) [reply docker/wiremock/stubs/__files/correctPatientNoDocsStructuredRecordResponse.json:247]: looks like it had the wrong display in GPC and has been fixed at some point https://termbrowser.nhs.uk/?perspective=full&conceptId1=886921000000105&edition=uk-edition&release=v20210317&server=https://termbrowser.nhs.uk/sct-browser-api/snomed&langRefset=999001261000000100,999000691000001104

Comment (ElliotB-96) [reply docker/wiremock/stubs/__files/correctPatientNoDocsStructuredRecordResponse.json:247]: Came from the public demonstrator update. @mattd-kainos do you think this is an issue?

Comment (mattd-kainos) [reply docker/wiremock/stubs/__files/correctPatientNoDocsStructuredRecordResponse.json:247]: Not an issue, that is the correct display for this snomed ct

**Review by mattd-kainos — APPROVED** [review]
